diff -ruN qtermwidget-1.4.0.dist/lib/qtermwidget.cpp qtermwidget-1.4.0/lib/qtermwidget.cpp
--- a/lib/qtermwidget.cpp	2024-01-18 16:18:15.609891396 -0500
+++ b/lib/qtermwidget.cpp	2024-01-19 09:26:12.302612778 -0500
@@ -331,6 +331,8 @@
             this, SIGNAL(termLostFocus()));
     connect(m_impl->m_terminalDisplay, &TerminalDisplay::keyPressedSignal, this,
             [this] (QKeyEvent* e, bool) { Q_EMIT termKeyPressed(e); });
+    connect(m_impl->m_terminalDisplay, &TerminalDisplay::mouseClickedSignal, this,
+            [this] (QMouseEvent* e) { Q_EMIT termMouseClicked(e); });
 //    m_impl->m_terminalDisplay->setSize(80, 40);
 
     QFont font = QApplication::font();
diff -ruN qtermwidget-1.4.0.dist/lib/qtermwidget.h qtermwidget-1.4.0/lib/qtermwidget.h
--- a/lib/qtermwidget.h	2024-01-18 16:18:15.609891396 -0500
+++ b/lib/qtermwidget.h	2024-01-19 09:20:17.919530634 -0500
@@ -267,6 +267,7 @@
     void termLostFocus();
 
     void termKeyPressed(QKeyEvent *);
+    void termMouseClicked(QMouseEvent *);
 
     void urlActivated(const QUrl&, bool fromContextMenu);
 
diff -ruN qtermwidget-1.4.0.dist/lib/TerminalDisplay.cpp qtermwidget-1.4.0/lib/TerminalDisplay.cpp
--- a/lib/TerminalDisplay.cpp	2024-01-18 16:18:15.607891329 -0500
+++ b/lib/TerminalDisplay.cpp	2024-01-19 09:31:28.437393293 -0500
@@ -1958,6 +1958,8 @@
     return;
   }
 
+  emit mouseClickedSignal(ev);
+
   if ( !contentsRect().contains(ev->pos()) ) return;
 
   if ( !_screenWindow ) return;
diff -ruN qtermwidget-1.4.0.dist/lib/TerminalDisplay.h qtermwidget-1.4.0/lib/TerminalDisplay.h
--- a/lib/TerminalDisplay.h	2024-01-18 16:18:15.607891329 -0500
+++ b/lib/TerminalDisplay.h	2024-01-19 09:25:59.037160407 -0500
@@ -534,6 +534,7 @@
      * Emitted when the user presses a key whilst the terminal widget has focus.
      */
     void keyPressedSignal(QKeyEvent *e, bool fromPaste);
+    void mouseClickedSignal(QMouseEvent *e);
 
     /**
      * A mouse event occurred.
